invenio_rdm_records/services/community_records/service.py

```diff
@@ -0,0 +1,119 @@
+
+from invenio_rdm_records.proxies import current_record_communities_service
+
+
+class CommunityRecordsService(RecordService):
+    """Community records service.
+
+    The record communities service is in charge of managing the records of a given community.
+    """
+
+    @property
+    def community_record_schema(self):
+        """Returns the community schema instance."""
+        return ServiceSchemaWrapper(self, schema=self.config.community_record_schema)
+
+    @property
+    def community_cls(self):
+        """Factory for creating a community class."""
+        return self.config.community_cls
+
+    def read(
+        self, identity, community_id, params=None, search_preference=None, **kwargs
+    ):
+        """Search for records published in the given community."""
+        self.require_permission(identity, "read")
+        self.community_cls.pid.resolve(community_id)  # Ensure community's existence
+
+        params = params or {}
+
+        search_result = self._search(
+            "search",
+            identity,
+            params,
+            search_preference,
+            record_cls=self.record_cls,
+            search_opts=self.config.search,
+            extra_filter=dsl.Q("term", **{"parent.communities.ids": str(community_id)}),
+            permission_action="read",
+            **kwargs,
+        ).execute()
+
+        return self.result_list(
+            self,
+            identity,
+            search_result,
+            params,
+            links_tpl=LinksTemplate(
+                self.config.links_search_community_records,
+                context={
+                    "args": params,
+                    "id": community_id,
+                },
+            ),
+            links_item_tpl=self.links_item_tpl,
+        )
+
+    def _remove(self, community, record, identity):
+        """Remove a community from the record."""
+        data = dict(communities=[dict(id=str(community.id))])
+        errors = current_record_communities_service.delete(
+            identity, record_id=record.pid.pid_value, data=data
+        )
+
+        return errors
+
+    @unit_of_work()
+    def delete(self, identity, community_id, data, revision_id=None, uow=None):
+        """Remove records from a community."""
+        community = self.community_cls.pid.resolve(community_id)
+        self.require_permission(identity, "delete_record", record=community)
+        valid_data, errors = self.community_record_schema.load(
+            data,
+            context={
+                "identity": identity,
+                "max_number": self.config.max_number_of_removals,
+            },
+            raise_errors=True,
+        )
+        records_dict = valid_data["records"]
+
+        for record_dict in records_dict:
+            record_id = record_dict["id"]
+            try:
+                record = self.record_cls.pid.resolve(record_id)
+                errors += self._remove(community, record, identity)
```

Comment:
Yes, it's generated when loading the data using the marshmallow schema and if there are no errors it's an empty list. If there are errors, since we have set the `raise_errors` to true it will fail with a 400.